osu.Server.Spectator/Hubs/MultiplayerHubContext.cs

```diff
@@ -128,26 +128,27 @@ public async Task ChangeAndBroadcastUserState(ServerMultiplayerRoom room, Multip
         {
             log(room, user, $"User state changed from {user.State} to {state}");
 
-            user.State = state;
-
             string? connectionId = users.GetConnectionIdForUser(user.UserID);
 
             if (connectionId != null)
             {
-                switch (state)
+                if (state == MultiplayerUserState.FinishedPlay)
                 {
-                    case MultiplayerUserState.FinishedPlay:
-                    case MultiplayerUserState.Idle:
-                        await context.Groups.RemoveFromGroupAsync(connectionId, MultiplayerHub.GetGroupId(room.RoomID, true));
-                        break;
-
-                    case MultiplayerUserState.Ready:
-                    case MultiplayerUserState.Spectating:
+                    // when entering FinishedPlay state, we can be confident that player can be removed from group
```

Comment:
I don't understand this comment.

Why can `FinishedPlay` be removed but not `Idle`, like in the old code? I guess this is a technical limitation from osu!'s side regarding transition from the results screen, which while if so, is probably fine for an initial implementation, the comment here just does nothing to explain anything.

Reply:
`Idle` player needs to be in gameplay group so that he will join gameplay when next match starts. The only time when player is not in a state fit for gameplay group is in `FinishedPlay` or `Results` states.

I will remove the comment.